File: app/src/test/java/com/farmerbb/taskbar/util/ToastFrameworkImplTest.java
package com.farmerbb.taskbar.util;

import android.content.Context;
import android.view.Gravity;
import android.widget.Toast;

import androidx.test.core.app.ApplicationProvider;

import com.farmerbb.taskbar.R;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.Shadows;
import org.robolectric.shadows.ShadowToast;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class ToastFrameworkImplTest {
    private Context context;
    private ToastFrameworkImpl impl;
    private String message = "test-message";
    private int length = Toast.LENGTH_LONG;

    @Before
    public void setUp() {
        context = ApplicationProvider.getApplicationContext();
        impl = new ToastFrameworkImpl(context, message, length);
    }

    @Test
    public void testShow() {
        impl.show();
        assertEquals(message, ShadowToast.getTextOfLatestToast());
        Toast toast = ShadowToast.getLatestToast();
        assertEquals(length, toast.getDuration());
        assertEquals(Gravity.BOTTOM | Gravity.CENTER_VERTICAL, toast.getGravity());
        assertEquals(0, toast.getXOffset());
        assertEquals(
                context.getResources().getDimensionPixelSize(R.dimen.tb_toast_y_offset),
                toast.getYOffset()
        );
    }

    @Test
    public void testCancel() {
        impl.show();
        ShadowToast toast = Shadows.shadowOf(ShadowToast.getLatestToast());
        assertFalse(toast.isCancelled());
        impl.cancel();
        assertTrue(toast.isCancelled());
    }
}

File: app/src/test/java/com/farmerbb/taskbar/util/ToastFrameworkImplTest.kt
package com.farmerbb.taskbar.util

import android.content.Context
import android.view.Gravity
import android.widget.Toast
import androidx.test.core.app.ApplicationProvider
import com.farmerbb.taskbar.R
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.Shadows
import org.robolectric.shadows.ShadowToast

@RunWith(RobolectricTestRunner::class)
class ToastFrameworkImplTest {
    private lateinit var context: Context
    private lateinit var impl: ToastFrameworkImpl
    private val message = "test-message"
    private val length = Toast.LENGTH_LONG

    @Before
    fun setUp() {
        context = ApplicationProvider.getApplicationContext()
        impl = ToastFrameworkImpl(context, message, length)
    }

    @Test
    fun testShow() {
        impl.show()
        Assert.assertEquals(message, ShadowToast.getTextOfLatestToast())
        val toast = ShadowToast.getLatestToast()
        Assert.assertEquals(length.toLong(), toast.duration.toLong())
        Assert.assertEquals((Gravity.BOTTOM or Gravity.CENTER_VERTICAL).toLong(),
                toast.gravity.toLong())
        Assert.assertEquals(0, toast.xOffset.toLong())
        Assert.assertEquals(
                context.resources.getDimensionPixelSize(R.dimen.tb_toast_y_offset).toLong(),
                toast.yOffset
                        .toLong())
    }

    @Test
    fun testCancel() {
        impl.show()
        val toast = Shadows.shadowOf(ShadowToast.getLatestToast())
        Assert.assertFalse(toast.isCancelled)
        impl.cancel()
        Assert.assertTrue(toast.isCancelled)
    }
}
